examples/tutorials/nb_python/Habitat_Interactive_Tasks.py

```diff
@@ -0,0 +1,2120 @@
+# @markdown - The simulator will render both RGB, Depth observations of 256x256 resolution.
+# @markdown - The actions available will be `move_forward`, `turn_left`, `turn_right`.
+
+
+def make_cfg(settings):
+    sim_cfg = habitat_sim.SimulatorConfiguration()
+    sim_cfg.gpu_device_id = 0
+    sim_cfg.default_agent_id = settings["default_agent_id"]
+    sim_cfg.scene.id = settings["scene"]
+    sim_cfg.enable_physics = settings["enable_physics"]
+    sim_cfg.physics_config_file = settings["physics_config_file"]
+
+    # Note: all sensors must have the same resolution
+    sensors = {
+        "rgb": {
+            "sensor_type": habitat_sim.SensorType.COLOR,
+            "resolution": [settings["height"], settings["width"]],
+            "position": [0.0, settings["sensor_height"], 0.0],
+        },
+        "depth": {
+            "sensor_type": habitat_sim.SensorType.DEPTH,
+            "resolution": [settings["height"], settings["width"]],
+            "position": [0.0, settings["sensor_height"], 0.0],
+        },
+    }
+
+    sensor_specs = []
+    for sensor_uuid, sensor_params in sensors.items():
+        if settings[sensor_uuid]:
+            sensor_spec = habitat_sim.SensorSpec()
+            sensor_spec.uuid = sensor_uuid
+            sensor_spec.sensor_type = sensor_params["sensor_type"]
+            sensor_spec.resolution = sensor_params["resolution"]
+            sensor_spec.position = sensor_params["position"]
+
+            sensor_specs.append(sensor_spec)
+
+    # Here you can specify the amount of displacement in a forward action and the turn angle
+    agent_cfg = habitat_sim.agent.AgentConfiguration()
+    agent_cfg.sensor_specifications = sensor_specs
+    agent_cfg.action_space = {
+        "move_forward": habitat_sim.agent.ActionSpec(
+            "move_forward", habitat_sim.agent.ActuationSpec(amount=0.1)
+        ),
+        "turn_left": habitat_sim.agent.ActionSpec(
+            "turn_left", habitat_sim.agent.ActuationSpec(amount=10.0)
+        ),
+        "turn_right": habitat_sim.agent.ActionSpec(
+            "turn_right", habitat_sim.agent.ActuationSpec(amount=10.0)
+        ),
+    }
+
+    return habitat_sim.Configuration(sim_cfg, [agent_cfg])
+
+
+settings = {
+    "max_frames": 10,
+    "width": 256,
+    "height": 256,
+    "scene": "data/scene_datasets/coda/coda.glb",
+    "default_agent_id": 0,
+    "sensor_height": 1.5,  # Height of sensors in meters
+    "rgb": True,  # RGB sensor
+    "depth": True,  # Depth sensor
+    "seed": 1,
+    "enable_physics": True,
+    "physics_config_file": "data/default.phys_scene_config.json",
+    "silent": False,
+    "compute_shortest_path": False,
+    "compute_action_shortest_path": False,
+    "save_png": True,
+}
+
+cfg = make_cfg(settings)
+
+
+# %%
+# @title Spawn the agent at a pre-defined location
+
+try:  # Got to make initialization idiot proof
+    sim.close()
+except NameError:
+    pass
+
+cfg.sim_cfg.default_agent_id = 0
+sim = habitat_sim.Simulator(cfg)
```

Comment:
This occurs to me what if you `with sim` over multiple cells, will it automatically close when it leaves scope, if so, this may be a clean way of coding it.